keras_nlp/samplers/beam_sampler.py

```diff
@@ -85,13 +86,26 @@ def get_next_token(self, next_token_probs):
         pass
 
     def sample(
-        self, prompt, token_probability_fn, mask, num_steps, from_logits=True
+        self,
+        prompt,
+        token_probability_fn,
+        mask,
+        num_steps,
+        from_logits=True,
+        cache=None,
+        token_probs=None,
     ):
         """Sampling logic implementation.
 
         Because beam search uses a different loop body, we have to override the
         whole `sample` method instead of just the `get_next_token` method.
         """
+        if cache is not None:
```

Comment:
How do we want to handle this? Should we just unexport beam for now, and leave cache stuff as a follow up?

Reply:
let's do a followup, and we can still export beam without cache.